runtime/jcl/common/shared.c

```diff
@@ -546,10 +547,24 @@ getPathProtocolFromURL(JNIEnv* env, jobject url, jmethodID URLgetPathID, jmethod
 	return rc;
 }
 
+static BOOLEAN
+isPathTypeJimage(const char *pathChars, jsize pathLen) {
+	char* endsWith = (char*)(pathChars + (pathLen-4));
```

Comment:
There is no need to declare `endsWith` here and set it to` (char*)(pathChars + (pathLen-4))`.  Line 557 can be changed to `char* endsWith = (char*)(pathChars + (pathLen - len));`